### PR TITLE
Grids: presetve repeat from inspector changes

### LINKED ISSUE #6387: Keep repeats when editing grid templates via the inspector

### PR DESCRIPTION
**Pull request by ruggi**
**Problem:**

As a followup to #6380 , we should preserve `repeat` functions & co. when altering the grid template from the inspector inputs too.

**Fix:**

1. Factored out the logic to alter a table from the resize strategy to helper functions in `grid-helpers.ts`
2. Adjusted the logic to support removing elements too
3. Refactored the callbacks in the inspector section to use the new helpers

Fixes #6387

### CODE REVIEW

File: editor/src/components/canvas/canvas-strategies/strategies/resize-grid-strategy.ts [hunk 1]
```diff
@@ -207,51 +190,6 @@ export const resizeGridStrategy: CanvasStrategyFactory = (
   }
 }
 
-type DimensionIndexes = {
```
Comment (ruggi) [inline editor/src/components/canvas/canvas-strategies/strategies/resize-grid-strategy.ts:210]: all this stuff now lives in `grid-hepers.ts`

**Review by seanparsons — APPROVED** [review]

File: editor/src/components/canvas/canvas-strategies/strategies/grid-helpers.ts [hunk 2]
```diff
@@ -647,3 +655,175 @@ function gridTemplateToNumbers(gridTemplate: GridTemplate | null): Array<number>
...
+export type ReplaceGridDimensionAction = {
+  type: 'REPLACE'
+  newValue: GridDimension
+}
+
+export type RemoveGridDimensionAction = {
+  type: 'REMOVE'
+}
+
+export type AlterGridTemplateDimensionAction =
```
Comment (gbalint) [inline editor/src/components/canvas/canvas-strategies/strategies/grid-helpers.ts:747]: I don't really like the Action terminology, we use that for our dispatchable actions, someone would easily think these actions are EditorActions too.

File: editor/src/components/canvas/canvas-strategies/strategies/grid-helpers.ts [hunk 3]
```diff
@@ -647,3 +655,175 @@ function gridTemplateToNumbers(gridTemplate: GridTemplate | null): Array<number>
...
+        expanded.push(...repeatGroup)
+      }
+      return [...acc, ...expanded]
+    } else {
+      return [...acc, expandedGridDimension(cur, index)]
+    }
+  }, [] as ExpandedGridDimension[])
+}
+
+function alterGridDimensions(params: {
```
Comment (gbalint) [inline editor/src/components/canvas/canvas-strategies/strategies/grid-helpers.ts:702]: This function does not generally alters the grid dimensions, but it alters the repeats, I would express this in the name

File: editor/src/components/canvas/canvas-strategies/strategies/grid-helpers.ts (hunk 2 above)
Comment (ruggi) [reply editor/src/components/canvas/canvas-strategies/strategies/grid-helpers.ts:747]: ah good point

File: editor/src/components/canvas/canvas-strategies/strategies/grid-helpers.ts (hunk 3 above)
Comment (ruggi) [reply editor/src/components/canvas/canvas-strategies/strategies/grid-helpers.ts:702]: it does alter them, you can replace or remove any dimension, not just repeats.

**Review by gbalint — APPROVED** [review]

**Pull request merged**